subsys/net/ip/net_context.c

```diff
@@ -816,6 +816,24 @@ NET_CONN_CB(tcp_established)
 				     sys_get_be32(NET_TCP_HDR(pkt)->ack));
 	}
 
+//printk("%u %u\n", sys_get_be32(NET_TCP_HDR(pkt)->seq), context->tcp->send_ack);
+
+	/* Peer sent us packet we've already seen. Apparently, our ack was lost. */
+	if (sys_get_be32(NET_TCP_HDR(pkt)->seq) < context->tcp->send_ack) {
+printk("Received already seen pkt - ACKing\n");
+
+	set_appdata_values(pkt, IPPROTO_TCP);
```

Comment:
This just parallels what's done in "normal ACK" branch: https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/ip/net_context.c#L827 . Apparently if I added it, it didn't work w/o it, but I don't remember exactly now.